.github/workflows/deploy.yml

```diff
@@ -1,88 +1,87 @@
-          push: true
-      -
-        name: Deploy server
-        uses: fjogeleit/http-request-action@v1.3.3
-        if: ${{ matrix.image == 'server' && github.event_name == 'push' }}
-        with:
-          url: 'https://rancher2.spin.nersc.gov/v3/project/c-tmq7p:p-bkv45/workloads/deployment:nmdc-dev:portal-backend?action=redeploy'
-          method: 'POST'
-          username: ${{ secrets.SPIN_USER }}
-          password: ${{ secrets.SPIN_PASSWORD }}
-#
-# CLIENT
-#
-      -
-        name: Build and push client
-        uses: docker/build-push-action@v2
-        if: ${{ matrix.image == 'client' }}
-        with:
-          tags: ${{ format('ghcr.io/microbiomedata/nmdc-server/client:{0}', github.event.release.tag_name || 'latest') }}
-          context: web
-          push: true
-      -
-        name: Deploy client
-        uses: fjogeleit/http-request-action@v1.3.3
-        if: ${{ matrix.image == 'client' && github.event_name == 'push' }}
-        with:
-          url: 'https://rancher2.spin.nersc.gov/v3/project/c-tmq7p:p-bkv45/workloads/deployment:nmdc-dev:portal-frontend?action=redeploy'
-          method: 'POST'
-          username: ${{ secrets.SPIN_USER }}
-          password: ${{ secrets.SPIN_PASSWORD }}
-#
-# WORKER
-#
-      -
-        name: Build and push worker
-        uses: docker/build-push-action@v2
-        if: ${{ matrix.image == 'worker' }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v5
         with:
-          tags: ${{ format('ghcr.io/microbiomedata/nmdc-server/worker:{0}', github.event.release.tag_name || 'latest') }}
-          file: ./Dockerfile.worker
-          push: true
-      - 
-        name: Deploy worker
-        uses: fjogeleit/http-request-action@v1.3.3
-        if: ${{ matrix.image == 'worker' && github.event_name == 'push' }}
+          context: ${{ matrix.image == 'client' && 'web' || '.' }}
+          push: ${{ env.IS_ORIGINAL_REPO }}
+          file: ${{ matrix.image == 'worker' && 'Dockerfile.worker' || (matrix.image == 'client' && 'web/Dockerfile' || 'Dockerfile') }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+  redeploy:
+    needs: build
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        deployment: [ backend, frontend, worker ]
+
+    steps:
+      - name: Redeploy ${{ env.RANCHER_NAMESPACE }}:${{ env.IS_PROD_RELEASE && '' || 'portal-' }}${{ matrix.deployment }}
+        if: ${{ env.IS_ORIGINAL_REPO }}
+        uses: fjogeleit/http-request-action@v1
         with:
-          url: 'https://rancher2.spin.nersc.gov/v3/project/c-tmq7p:p-bkv45/workloads/deployment:nmdc-dev:portal-worker?action=redeploy'
-          method: 'POST'
+          url: https://rancher2.spin.nersc.gov/v3/project/c-tmq7p:p-bkv45/workloads/deployment:${{ env.RANCHER_NAMESPACE }}:${{ env.IS_PROD_RELEASE && '' || 'portal-' }}${{ matrix.deployment }}?action=redeploy
```

Comment:
I see that this is where the prefix of `portal-` would be applied when needed (i.e. in the `nmdc-dev` namespace).

Reply:
Cool thanks. It'll be nice when they're all consistent, but no particular rush. We'll just have to remember to update this workflow again once that issue is resolved.